rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMController.java

```diff
@@ -867,37 +910,42 @@ public void onSuccess(String result) {
             JSONObject p = arr.get(i).isObject();
 
             String pluginName = p.get("pluginName").isString().stringValue();
-            String pluginDescription = p.get("pluginDescription").isString().stringValue();
-            PluginDescriptor desc = new PluginDescriptor(pluginName, pluginDescription);
+            PluginDescriptor desc = getPluginDescriptor(p, pluginName);
 
-            JSONArray fields = p.get("configurableFields").isArray();
-            for (int j = 0; j < fields.size(); j++) {
-                JSONObject field = fields.get(j).isObject();
+            plugins.put(pluginName, desc);
+        }
 
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)